packages/generate/lib/github.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const { Octokit } = require( '@octokit/rest' );
+const { RequestError } = require( '@octokit/request-error' );
+const { name: packageName, version: packageVersion } = require( '../package.json' );
+const { sleep } = require( './utils' );
+
+const github = {};
+
+let octokit;
+
+github.initialize = ( token ) => {
+	octokit = new Octokit( {
+		auth: token,
+		userAgent: `${ packageName }/${ packageVersion }`,
+		previews: [ 'baptiste' ],
+	} );
+};
+
+github.createRepositoryUsingTemplate = async ( {
+	templateOwner,
+	templateName,
+	owner,
+	name,
+	isPrivate,
+	description,
+} ) => {
+	const response = await octokit.repos.createUsingTemplate( {
+		template_owner: templateOwner,
+		template_repo: templateName,
+		owner,
+		name,
+		private: isPrivate,
+		description,
+		include_all_branches: false,
+	} );
+
+	return response.data;
+};
+
+github.getRepository = async ( owner, name ) => {
+	const response = await octokit.repos.get( { owner, repo: name } );
+	return response.data;
+};
+
+github.hasRepository = async ( owner, name ) => {
+	try {
+		await github.getRepository( owner, name );
+		return true;
+	} catch ( error ) {
+		if ( error instanceof RequestError && 404 === error.status ) {
+			return false;
+		}
+
+		throw error;
+	}
+};
+
+github.waitUntilRepositoryIsReady = async ( owner, name ) => {
+	const retryDelays = [ 0.5, 1, 1, 2, 3, 4, 5 ].map( ( a ) => a * 1000 );
```

Comment:
Wieso gibt es zwei mal `1`?

Reply:
It's just a random next delay. It starts slowly and ends at a check every 5 seconds.